File: nuxt-client/app/types/PostSummary.ts
export default interface PostSummary {
  id: number;
  content: string;
  createdAt: string;
  authorId: string;
  authorName: string;
  postCount: number;
  likeCount: number;
  saveCount: number;
  urls: string;
  imageUrls: string[];
  rootPostId?: number;
  postId?: number;
}

File: backend/post/src/main/java/com/backend/controller/dto/PostSummary.java
package com.backend.controller.dto;

import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PostSummary {

  private Integer id;
  private String content;
  private Instant createdAt;
  private Integer authorId;
  private String authorName;
  private Integer postCount;
  private Integer likeCount;
  private Integer saveCount;
  private String tags;
  private String urls;
}
